rfc/graphql-schema/additionalTypeDefs/companyRegistration.ts

```diff
@@ -0,0 +1,69 @@
+export default /* GraphQL */ `
+  enum CompanyStatus {
+    PENDING
+    APPROVED
+    REJECTED
+    INACTIVE
+    DELETED
+  }
+
+  extend type Company {
+    id: ID!
+    status: CompanyStatus!
+  }
+
+  input AddCompanyUsersInput {
+    extraFields: CompanyExtraFieldsInput
+  }
+
+  input AddCompanyFileInput {
+    id: ID!
+  }
+
+  input CompanyExtraFieldsInput {
+    multipleChoices: [MultipleChoiceFormFieldInput!]
+    numbers: [NumberFormFieldInput!]
+    texts: [TextFormFieldInput!]
+    multilineTexts: [MultilineTextFormFieldInput!]
+  }
+
+  input AddCompanyAddressInput {
+    firstName: String!
+    lastName: String!
+    address1: String!
+    address2: String
+    city: String!
+    countryCode: String!
+    stateOrProvince: String
+    phone: String
+    postalCode: String
+    extraFields: CompanyExtraFieldsInput
+  }
+
+  input RegisterCompanyInput {
+    name: String!
+    email: String!
+    phone: String!
+    # What's this for? doesn't seem to show in the UI
+    taxExemptCategory: String
+    fileList: [AddCompanyFileInput!]
+    companyUser: AddCompanyUsersInput
+    extraFields: CompanyExtraFieldsInput
+    address: AddCompanyAddressInput!
+  }
+
+  union RegisterCompanyError = ValidationError
+
+  type RegisterCompanyResult {
+    company: Company
+    errors: [RegisterCompanyError!]!
+  }
+
```

Comment:
We also need to surface the type of Registration flows allowed, when b2b features are enabled - the merchant has the ability to allow the following scenarios:
- B2C (Customer Registration Only - No Company Registration allowed)
- B2B (Company Registration Required - if only having Customer Entity - force the user to sign up for Company)
- B2C & B2B

Based on conversations already, this would best be modeled as an ENUM instead of two booleans because the booleans would allow you to build a logic for neither being allowed which is invalid state.

Based on reading the existing Site settings query, we could structure this inside the `CustomerSettings` called `RegistrationTypesAllowed`.

Any thoughts on this? @bigcommerce/team-b2b

Reply:
added some draft schema for this. Please have a look at the most recent commit.